### PR TITLE
PENG-4459

### PR DESCRIPTION
**Pull request by shreyasgune**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            def mycolor = 'good'
+            sites.each { site ->
+              sh "curl -o ${site}-latest.txt https://www.${site}.com/robots.txt"
+              def identical = sh(script: "diff -q -B ${site}-latest.txt seo-robots/${site}.txt", returnStatus: true) == 0
+              if (!identical) {
+                mycolor = 'danger'
+                def check_robots = sh(script: "set +e; diff -s -B ${site}-latest.txt seo-robots/${site}.txt; true", returnStdout: true).trim()
+                echo "${site} differed:\n${check_robots}"
+                currentBuild.result = 'UNSTABLE'
+                if("${site}"=="gametimepa"){
```
Comment (kdw174) [inline Jenkinsfile]: what is different about gametimepa?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                }else {
+                  slackSend color: 'danger',
+                            channel: '#seo-robots-check',
+                            message: "Processed robots.txt for ${site} and found differences: ```${check_robots}```"
+                }
+              }
+            }
+            slackSend color: "${mycolor}",
+                      channel: '#seo-robots-check',
+                      message: "Hey @here ! robots.txt check has been run and succesfully completed for all uscp sites. Discrepancies (if any) have been listed above."
```
Comment (kdw174) [inline Jenkinsfile]: I don't think that @ here will work. You have to format it some specific way to get it to work in a webhook. qe can probably help or we can get rid of the at here all together. We probably don't need to do an @ here if there weren't any issues.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          }catch(err){
+            slackSend color: 'danger',
+                      channel: '#seo-robots-check',
+                      message: "Hey @here! robots.txt check has resulted in failure. Please contact `#sre` to troubleshoot the jenkins Cron job. ```${err}```"
+          }
+        }
+      }
+      post {
+        cleanup {
+          sh 'rm -r *-latest.txt || true'
```
Comment (kdw174) [inline Jenkinsfile]: -r is for a directory, probably should be -f

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+              }
+            }
+            slackSend color: "${mycolor}",
+                      channel: '#seo-robots-check',
+                      message: "Hey @here ! robots.txt check has been run and succesfully completed for all uscp sites. Discrepancies (if any) have been listed above."
+
+          }catch(err){
+            slackSend color: 'danger',
+                      channel: '#seo-robots-check',
+                      message: "Hey @here! robots.txt check has resulted in failure. Please contact `#sre` to troubleshoot the jenkins Cron job. ```${err}```"
```
Comment (kdw174) [inline Jenkinsfile]: Same thing with this @ here

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        }
+      }
+      post {
+        cleanup {
+          sh 'rm -r *-latest.txt || true'
+        }
+      }
+    }
+  }
+}
```
Comment (kdw174) [inline Jenkinsfile]: Do we want to put this in a kube cronjob instead or is jenkins acceptable for this?

File: Jenkinsfile (hunk 2 above)
Comment (shreyasgune) [reply Jenkinsfile]: I tested it out and it worked on that channel. Sutton said he wants everyone on the channel to get notified that the job did run and of any issues that happen.

File: Jenkinsfile (hunk 5 above)
Comment (shreyasgune) [reply Jenkinsfile]: Lindeman suggested this be a jenkins job so I just ran with it.

**Comment by rgindes** [comment]
@kdw174 the stuff you brought up here is valid but i'm cool with jamming this through and revisiting in the future as time allows -- kyle sutton wants to be at-here'd, it's a pretty low volume channel (just the three of us right now) so i'm cool with leaving it. we can always take it out if ten million people join the channel and get mad at us for spamming them.

we can talk about moving to a kube cron job in the future as well, probably the correct long term home but if QE is cool with this living in jenkins then it passes my test.

**Pull request merged**